### ISSUE TITLE
Improving PPO

### ISSUE THREAD

**Issue by TolgaOk** [TolgaOk/Modular-Baselines#9]
## [Reproducing the official PPO implementation](https://ppo-details.cleanrl.dev//2021/11/05/ppo-implementation-details/)

Please check the link above for details.

- 13 core implementation details
- 9 Atari specific implementation details
- 9 implementation details for robotics tasks (with continuous action spaces)
- 5 LSTM implementation details
- 1 MultiDiscrete implementation detail

### Core implementation details
- [x] Vectorized architecture
- [ ] Orthogonal Initialization of Weights and Constant Initialization of biases
  - The policy output layer weights are initialized with the scale of 0.01. The value output layer weights are initialized with the scale of 1
  - [torch.nn.init.orthogonal_](https://pytorch.org/docs/stable/_modules/torch/nn/init.html#orthogonal_)
- [ ] The Adam Optimizer’s Epsilon Parameter
  - PPO sets the epsilon parameter to 1e-5
- [ ] Adam Learning Rate Annealing
  - In MuJoCo, the learning rate linearly decays from 3e-4 to 0
  - Atari games set the learning rate to linearly decay from 2.5e-4 to 0
- [x] Generalized Advantage Estimation
  - Termination caused by environment length limits must be count as non terminals in target value calculation
- [x] Mini-batch Updates
- [ ] Normalization of Advantages
  - After calculating the advantages based on GAE, PPO normalizes the advantages by subtracting their mean and dividing them by their standard deviation. In particular, this normalization happens at the minibatch level instead of the whole batch level!
- [ ] Value Function Loss Clipping
- [x] Overall Loss and Entropy Bonus
- [x] Global Gradient Clipping
- [ ] Debug variables
  - policy_loss
  - value_loss
  - entropy_loss
  - clipfrac
  - approxkl
- [ ] Shared and separate MLP networks for policy and value functions

### 9 details for continuous action domains (e.g. Mujoco)

- [x] Continuous actions via normal distributions
- [ ] State-independent log standard deviation
- [x] Independent action components
- [ ] Separate MLP networks for policy and value functions
- [ ] Handling of action clipping to valid range and storage
   - The original unclipped action is stored as part of the episodic data
- [ ]
